core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -693,7 +694,14 @@ public PlanOptimizers(
                         statsCalculator,
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(new RemoveRedundantIdentityProjections())),
-
+                // Prefer write partitioning rule requires accurate stats.
+                // Therefore PredicatePushDown, columnPruningOptimizer and
+                // RemoveRedundantIdentityProjections need to run beforehand.
```

Comment:
nit: we could remove `RemoveRedundantIdentityProjections` from both comments (here and below) as it should not affect stats. This could be separate PR

Reply:
I'll handle it when I get some free cycles